modules/library/http/src/main/java/org/geotools/http/AbstractHttpClient.java

```diff
@@ -90,6 +105,23 @@ public boolean isTryGzip() {
         return tryGzip;
     }
 
+    protected static URL appendURL(URL oldUrl, Map<String, Object> appendQuery)
+            throws MalformedURLException {
+        String oldQuery = oldUrl.getQuery();
+
+        StringJoiner stringJoiner = new StringJoiner("&");
+        appendQuery.forEach((key, value) -> stringJoiner.add(key + "=" + value));
```

Comment:
Don't we need URL encoding here, for both key and value?

Reply:
Done